Compiler cog

anthonyrys opened the pull request.

ramonechen committed:
Update globals

Added a resource path function for possible future builds of the project

Commit:
Add compile command

Commit:
Update compiler

anthonyrys committed:
Merge branch 'main' into compiler-cog

Commit:
Update compiler to fix C/CPP errors

Commit:
Merge branch 'compiler-cog' of https://github.com/SameriteRL/CARPI-Bot into compiler-cog

Commit:
Update compiler to fix more C/CPP errors

Commit:
Update compiler error handler

ramonechen reviewed (changes requested):
Overall I think this code is really well done. One definite but small issue is that `sendDefaultError()` has since been renamed `sendUnknownError()` in the main branch, and now takes in the error as a second positional arg. `TOKEN.txt` and `PREFIX.txt` are now in `src/assets` as well. Just update with main, make sure code runs fine afterwards and you're chill.

The rest of this stuff is just nitpicking and is up to you whether to leave in:
- I think type hinting variable assignments like `compile_request: requests.Response` and `language: str` are a bit redundant. I feel like it's easy enough to see what data type is being assigned to it and as long as the variable name is descriptive.
- `list.split()` splits by a space character by default, no need to specify `' '` in `args.split(' ')`.
- You can use an alias like `CompileLanguageError = commands.UserInputError` in the global space instead of creating an empty class.
- Not sure if `requireTwoArgumentsError()` or `languageNotAvailableError()` are beneficial as functions unless you plan on reusing them for future commands or if they just make things neater.

anthonyrys committed:
Merge pull request #5 from SameriteRL/main

Merge pull request from main

Commit:
Add requested changes, Update character replacement

Commit:
Update compiler

- Add asyncio implementation
- Add help embed
- Add positional arguments

ramonechen reviewed (changes requested):
Looks pretty good! Just a question, have you gotten this function to work as a slash command? It doesn't seem to work when I try it; if it's not doable as a slash command I'd recommend changing the decorator from `@commands.hybrid_command()` to `@commands.command()`.

Another few small things:
- Remove the redundant imports (asyncio, typing, requests)
- No need for `languageNotAvailableError()` to take in an `error` parameter if it won't use it
- Include `error` as a second argument to `sendUnknownArgument()`

After these changes, I think this is pretty much ready to merge with main :)

Commit:
Update compiler

SameriteRL commented:
I think this is good enough to merge with main! For future commits, maybe make some error messages more comprehensive; in the "language not available error", maybe list out the languages that are available. I know they're already included in the default help message, but make things as convenient and intuitive for the user as possible. There was also an error I observed when trying to run infinite loops in the online compiler, it raises an error like this:

`Command "compile" in cog "Compiler": Command raised an exception: HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
In embeds: Embed size exceeds maximum size of 6000`

Not sure how easy that is to fix, this is maybe also something you would want to create a custom error message for. I don't have much to say other than that; you can continue ironing out this cog, add support for more languages, or bring me some new ideas for the bot that you'd like to work on :)

anthonyrys committed:
Merge pull request #9 from SameriteRL/main

Merge from main

anthonyrys committed:
Merge pull request #12 from SameriteRL/main

Merge changes

Commit:
Update compiler

- Update code with PEP 8 guidelines
- Add code block decorator compatibility
- Add truncation for large outputs
